output/DeviceListAuditTool.py

```diff
@@ -0,0 +1,188 @@
+import pandas as pd
+import urllib
+import time
+import datetime
+from dateutil import parser as dtpars
+from urllib.request import urlopen
+import argparse
+from pathlib import Path
+from io import StringIO
+
+### Make a CLI argument parser
+parser = argparse.ArgumentParser(description="usage: %prog [options] <haystack.txt> --needlefile \n")
+### Add options
+parser.add_argument ('haystackfile', default='linac_logger_drf_requests.txt',
+                     help="[Path and] filename for DRF requests. (default: linac_logger_drf_requests.txt)")
+parser.add_argument ('-v', dest='debug', action="store_true", default=False,
+                     help="Turn on verbose debugging. (default: False)")
+parser.add_argument ('--maxcount',  dest='maxcount', default=-1,
+                     help="Number of devices in list to process. (default: -1 = all)")
+parser.add_argument ('--outdir',  dest='outdir', default='',
+                     help="Directory to write final output file. (default: pwd)")
+parser.add_argument ('--needlefile',  dest='needle_filename', default='',
+                     help="[Path and] file name of needles to be found in DRF requests (default: '')")
```

Comment:
I like the name. I think this description could better describe it's purpose. After reading the code, I think these are known problematic devices.

Reply:
Changed to `help="[Path and] file name of known-problematic DRF requests (default: '')"`